embedded/remotestorage/s3/s3.go

```diff
@@ -732,39 +735,39 @@ func (s *Storage) getRoleCredentials() error {
 		for {
 			select {
 			case _ = <-s3CredentialsRefreshTicker.C:
-				accessKeyID, secretKey, err := s.requestCredentials()
-				if err != nil {
-					log.Printf("S3 role credentials lookup failed with an error: %v", err)
-					continue
-				}
-				s.accessKeyID, s.secretKey = accessKeyID, secretKey
+					accessKeyID, secretKey, sessionToken, err := s.requestCredentials()
+					if err != nil {
+						log.Printf("S3 role credentials lookup failed with an error: %v", err)
+						continue
+					}
+					s.accessKeyID, s.secretKey, s.sessionToken = accessKeyID, secretKey, sessionToken
 			}
 		}
 	}()
 
 	return nil
 }
 
-func (s *Storage) requestCredentials() (string, string, error) {
+func (s *Storage) requestCredentials() (string, string, string, error) {
 	tokenReq, err := http.NewRequest("PUT", fmt.Sprintf("%s%s",
 		s.awsInstanceMetadataURL,
 		"/latest/api/token",
 	), nil)
 	if err != nil {
-		return "", "", errors.New("cannot form metadata token request")
+		return "", "", "", errors.New("cannot form metadata token request")
 	}
 
 	tokenReq.Header.Set("X-aws-ec2-metadata-token-ttl-seconds", "21600")
```

Comment:
I think we should make this configurable or at least move it to a `const`

Reply:
Actually, in this case the access_key, secret_key and session_token are taken from the `GET /latest/api/token` made in that function. And such credentials are temporary, so I think that for the role we can't configure the session_token and it can't be a const.

Reply:
@lorepas: I'm talking about `X-aws-ec2-metadata-token-ttl-seconds`